sound/soc/soc-dai.c

```diff
@@ -391,6 +391,45 @@ bool snd_soc_dai_stream_valid(struct snd_soc_dai *dai, int dir)
 	return stream->channels_min;
 }
 
+/*
+ * snd_soc_dai_link_set_capabilities() - set dai_link properties based on its DAIs
+ */
+void snd_soc_dai_link_set_capabilities(struct snd_soc_dai_link *dai_link)
+{
+	struct snd_soc_dai_link_component *cpu;
+	struct snd_soc_dai_link_component *codec;
+	struct snd_soc_dai *dai;
+	bool supported_direction;
+	int direction;
+	int i;
+
+	for_each_pcm_streams(direction) {
+		supported_direction = true;
+
+		for_each_link_cpus(dai_link, i, cpu) {
+			dai = snd_soc_find_dai(cpu);
+			if (!dai || !snd_soc_dai_stream_valid(dai, direction)) {
+				supported_direction = false;
+				break;
+			}
+		}
+		for_each_link_codecs(dai_link, i, codec) {
```

Comment:
you don't need to run this second loop at all if `supported_direction` is already `false`

Reply:
added this optimization but a goto would sure help here....